### DISCUSSION TITLE
How to import database structure?

### DISCUSSION THREAD

**Discussion by mbnoimi** [pocketbase/pocketbase#1692] [Q&A] [answered]
We use in our company a huge MySQL database (600 tables and tens of fields). I converted it to SQLite successfully today.

Is there any quick way to convert this database into PocketBase database? 
OR at least import the structure and/or data (it's about 90 GB) into a running PocketBase database? 

BTW, Our database deals with a huge amount of requests. I'm try to replace Laravel/MySQL REST API with PocketBase (I know this is a complete madness but I believe SQLite able to deal with it)

**Comment by mbnoimi** [comment]
I'm sorry. I post it in issue tab. Silly me

**Comment by ganigeorgiev** [comment, marked as answer]
> Is there any quick way to convert this database into PocketBase database?

No, unfortunately at the moment there is no "autodiscovery"-like type of import (_there are plans to implement it eventually sometime in the future but [other tasks](https://github.com/orgs/pocketbase/projects/2) for now are with higher priority_).

If you want to import external database, you'll have to:

1. Create a `_collections` entry for each of your tables.

2. Make sure that each record in your tables has the system fields required by their [collection type](https://pocketbase.io/docs/collections/) (eg. `id`, `created`, `updated`). Additionally, you have to make sure that datetimes, file fields, etc. comply with the field format that PocketBase is using (for example, dates are stored in `Y-m-d H:i:s.uZ` string format; file fields are stored by keeping only the filename and the files are expected to be found in `collectionId/recordId` directory/prefix)

**Reply by mbnoimi** [reply]
OMG, Bad news. 
I'll try to write a some script to fix this issue. Because it's impossible to recreating the structure using the web interface.

**Reply by ganigeorgiev** [reply]
You can find some examples how to create collections programmatically with Go in https://pocketbase.io/docs/collection-methods/#create-new-collection-without-data-validations.

The above is not required and you should be able to also execute `INSERT` statements directly against the `_collections` table, but please note that the `schema` column is a serialized json array where each field have the following fields:

```js
{
    name: 'title',
    type: 'text', // text, number, bool, email, url, date, select, json, file, relation
    required: true,
    unique: false,
    options: {
        min: "10",
        // all option fields you can find in:
        // https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.11.4/models/schema
        // https://github.com/pocketbase/pocketbase/blob/2128b155414ea6eee8615f388c284f45bbbb10cf/models/schema/schema_field.go#L368-L580
    },
},
```

**Comment by mbnoimi** [comment]
Thank you @ganigeorgiev
